### PR TITLE
stgit: 1.1 -> 1.3

### PR DESCRIPTION
**Pull request by terinjokes**
###### Motivation for this change

Two new updates.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by figsoda** [comment]
can you change your commit message to `stgit: 1.1 -> 1.3`?
`gitAndTools.*` were moved to top level in #109396

**Comment by terinjokes** [comment]
Oh, wow. I've never noticed that, and there's been several PRs updating stgit since. 😅

Yeah, can change it. One moment please.

File: pkgs/applications/version-management/git-and-tools/stgit/default.nix [hunk 1]
```diff
@@ -4,23 +4,25 @@
...
     owner = "stacked-git";
     repo = "stgit";
     rev = "v${version}";
-    sha256 = "sha256-gfPf1yRmx1Mn1TyCBWmjQJBgXLlZrDcew32C9o6uNYk=";
+    sha256 = "0wa3ba7afnbb1h08n9xr0cqsg93rx0qd9jv8a34mmpp0lpijmjw6";
   };
 
-  nativeBuildInputs = [ installShellFiles asciidoc xmlto docbook_xsl ];
+  nativeBuildInputs =
+    [ installShellFiles asciidoc xmlto docbook_xsl docbook_xml_dtd_45 ];
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/git-and-tools/stgit/default.nix:25]:
```suggestion
  nativeBuildInputs = [ installShellFiles asciidoc xmlto docbook_xsl docbook_xml_dtd_45 ];
```

File: pkgs/applications/version-management/git-and-tools/stgit/default.nix [hunk 2]
```diff
@@ -34,6 +36,14 @@ python3Packages.buildPythonApplication rec {
...
                   ${docbook_xsl}/xml/xsl/docbook/html/docbook.xsl
     done
+
+    substituteInPlace Documentation/texi.xsl \
+      --replace http://www.oasis-open.org/docbook/xml/4.5/docbookx.dtd \
+                ${docbook_xml_dtd_45}/xml/dtd/docbook/docbookx.dtd
+
+    cat>stgit/_version.py<<EOF
+    __version__ = "${version}"
+    EOF
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/git-and-tools/stgit/default.nix:46]:
```suggestion
    cat >stgit/_version.py <<EOF
    __version__ = "${version}"
    EOF
```

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 139688` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>stgit</li>
  </ul>
</details>

**Review by jshholland — APPROVED** [review]
All LGTM.

Result of `nixpkgs-review pr 139688` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>stgit</li>
  </ul>
</details>

**Pull request merged**